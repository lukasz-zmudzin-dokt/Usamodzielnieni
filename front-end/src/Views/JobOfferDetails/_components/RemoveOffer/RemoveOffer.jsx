import React, { useState, useRef, useContext } from "react";
import { Alert, Button } from "react-bootstrap";
import { deleteOffer } from "../../functions/deleteOffer";
import { AlertContext } from "context";

const RemoveOffer = ({ id, user }) => {
  const [confirmDeletion, setConfirmDeletion] = useState(false);
  const alertC = useRef(useContext(AlertContext));

  const handleDeleteOffer = async () => {
    try {
      await deleteOffer(id, user.token);
      alertC.current.showAlert("Pomyślnie usunięto ofertę.", "success");
    } catch (err) {
      alertC.current.showAlert("Wystąpił błąd przy usuwaniu oferty.");
    }

    setConfirmDeletion(false);
  };

  const msg = confirmDeletion ? (
    <Alert variant="warning">
      Czy na pewno chcesz usunąć tę ofertę?
      <Button variant="warning" className="ml-3" onClick={handleDeleteOffer}>
        Tak
      </Button>
    </Alert>
  ) : null;

  return (
    <div className="removeOffer">
      {msg || (
        <Button variant="danger" onClick={(e) => setConfirmDeletion(true)}>
          Usuń ofertę
        </Button>
      )}
    </div>
  );
};

export default RemoveOffer;
