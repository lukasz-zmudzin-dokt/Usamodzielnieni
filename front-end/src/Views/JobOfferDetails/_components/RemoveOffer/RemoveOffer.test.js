import React from "react";
import { render, waitForElement } from "@testing-library/react";
import RemoveOffer from "./RemoveOffer";

let mock_apiError = false;
jest.mock('../../functions/deleteOffer', () => ({ 
    deleteOffer: (id, token) => {
        if (mock_apiError) {
            throw Error('err');
        }
        return;
    }
}));

describe('RemoveOffer', () => {
    let props;

    beforeAll(() => {
        props = {
            id: '123',
            user: { token: 'abc' }
        };
    });

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should render without crashing', () => {
        const { container } = render(<RemoveOffer {...props}/>);

        expect(container).toMatchSnapshot();
    });

});