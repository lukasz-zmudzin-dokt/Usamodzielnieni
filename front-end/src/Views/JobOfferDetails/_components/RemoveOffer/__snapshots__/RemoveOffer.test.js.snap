// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveOffer should render without crashing 1`] = `
<div>
  <div
    class="removeOffer"
  >
    <button
      class="btn btn-danger"
      type="button"
    >
      Usuń ofertę
    </button>
  </div>
</div>
`;
